### PR TITLE
Add a new GetRelatedLocations API which uses the new cache keyed by stat var DCID.

### PR DESCRIPTION
**Pull request by Spaceenter**

### CODE REVIEW

File: server/places.go [hunk 1]
```diff
@@ -169,6 +169,79 @@ func (s *Server) GetRelatedPlaces(ctx context.Context,
...
+	}
+	if !util.CheckValidDCIDs([]string{in.GetDcid()}) {
+		return nil, fmt.Errorf("invalid DCID")
+	}
+
+	withinPlace := in.GetWithinPlace()
+	samePlaceType := in.GetSamePlaceType()
+	isPerCapita := in.GetIsPerCapita()
+	var prefix string
+	if withinPlace == "" {
```
Comment (shifucun) [inline server/places.go:186]: wonder if it's better to have a map from <bool, bool, bool> to prefix in here?

Comment (Spaceenter) [reply server/places.go]: Good point - done.

File: server/bigtable.go [hunk 2]
```diff
@@ -17,19 +17,31 @@ package server
...
+)
+
 // bigTableReadRowsParallel reads BigTable rows in parallel,
 // considering the size limit for RowSet is 500KB.
 func bigTableReadRowsParallel(
 	ctx context.Context,
 	btTable *bigtable.Table,
 	rowSet bigtable.RowSet,
 	action func(string, []byte) (interface{}, error),
+	tokenType TokenType,
```
Comment (shifucun) [inline server/bigtable.go:44]: i think it does the same thing as the opts param below where i call RemoveKeyPrefix(), so can remove opts?

Comment (shifucun) [inline server/bigtable.go:44]: do you think it better to pass the "extract token" function directly in here instead my old way of a bool or a enum here?

Comment (Spaceenter) [reply server/bigtable.go]: Done.

Comment (Spaceenter) [reply server/bigtable.go]: Done.

**Comment by shifucun** [comment]
nice to have getToken()!

**Pull request merged**